projects/virtual-machine-service-typescript/src/hard-coded-contracts/registry/bar-smart-contract.ts

```diff
@@ -0,0 +1,48 @@
+import BaseSmartContract from "../base-smart-contact";
+
+export default class BarSmartContract extends BaseSmartContract {
+    static readonly SUPERUSER = "0000";
+
+    public async transfer(recipient: string, amount: number) {
+        if (amount === 0) {
+            throw new Error("transaction amount must be > 0");
+        }
+
+        const senderBalance: number = await this.getBalance(this.sender);
```

Comment:
We should start using `BigNumber` only, in these cases. The token amounts and denominations we will be working with won't fit the `number` resolution.

Reply:
if BigNumber is like Java's BigDecimal, then it's also mandatory due to precision and rounding issues

Reply:
I agree. IMHO, we should switch to use only BigNumber's before it'd be too late.

Reply:
We're not using BigNumber (yet) since we need _protobuf_ and _grpc_ to support it. There's a separate task on that and it will hopefully be solved in the next few days. I've already started exploring it.